pkg/controller/installation/products/fuse/reconciler.go

```diff
@@ -0,0 +1,212 @@
+package fuse
+
+import (
+	"context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/marketplace"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/products/config"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	v1alpha12 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	errors2 "k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+    syn "github.com/syndesisio/syndesis/install/operator/pkg/apis/syndesis/v1alpha1"
+	"time"
+)
+
+const (
+	defaultInstallationNamespace = "fuse"
+	defaultSubscriptionName      = "syndesis"
+)
+
+type Reconciler struct {
+	client        pkgclient.Client
+	coreClient    *kubernetes.Clientset
+	Config        *config.Fuse
+	ConfigManager config.ConfigReadWriter
+	mpm           marketplace.MarketplaceInterface
+	logger        *logrus.Entry
+	nsReconciler  *resources.NamespaceReconciler
+}
+
+func NewReconciler(client pkgclient.Client, rc *rest.Config, coreClient *kubernetes.Clientset, configManager config.ConfigReadWriter, instance *v1alpha1.Installation, logger *logrus.Entry) (*Reconciler, error) {
+	fuseConfig, err := configManager.ReadFuse()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrieve keycloak codeReadyConfig")
+	}
+
+	if fuseConfig.GetNamespace() == "" {
+		fuseConfig.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+	if err = fuseConfig.Validate(); err != nil {
+		return nil, errors.Wrap(err, "keycloak codeReadyConfig is not valid")
+	}
+
+	mpm := marketplace.NewManager(client, rc)
```

Comment:
This has to be passed in now, otherwise we can't mock this mpm instance. You can see how I've done it in my unit test PR.